src/main/scala/org/deepdive/ddlog/DeepDiveLogSemanticChecker.scala

```diff
@@ -36,26 +36,27 @@ object DeepDiveLogSemanticChecker extends DeepDiveLogHandler {
     checkNumberOfColumns(stmt)
   }
 
+  // iterate over all atoms contained in the body list and apply the checker
+  def checkBodyAtoms(bodies: List[Body], checker: Atom => Unit) : Unit = {
```

Comment:
Let's [curry](http://www.scala-lang.org/node/135) this function to make it easier to use:
```scala
def checkBodyAtoms(checker: Atom => Unit)(bodies: List[Body]) : Unit = {
```
Then later in any real checker that matches cases for statements, etc. can be simplified.  For example `checkRelationDefined` below can become:
```scala
val check = checkBodyAtoms(checkRelation)
...
stmt match {
  case s: ExtractionRule => s.q.bodies foreach check
  case s: InferenceRule => s.q.bodies foreach check
  ...
}
```